core/src/main/scala/reftree/diagram/Animation.scala

```diff
@@ -83,6 +83,32 @@ object Animation {
   /** Create an animation builder from a sequence of starting values */
   def startWithSequence[A: ToRefTree](start: Seq[A]) = Builder(start.toVector)
 
+  private implicit class TakeUntilStream[T](s: Stream[T]) {
+    def takeWhileInclusive(p: T => Boolean): Stream[T] =
+      if (!s.isEmpty && p(s.head)) Stream.cons(s.head, s.tail takeWhileInclusive p)
+      else Stream.cons(s.tail.head, Stream.empty)
+  }
+
+  /** Create an animation builder from a seed until a fixpoint (f(x) == x) is reached */
+  def iterateUntilFixPoint[A: ToRefTree](seed: A)(iteration: (A => A))(
+    max: Int = 10) =
```

Comment:
I suggest to reorder the arguments a bit:
* `def iterateUntilFixpoint(iteration: A ⇒ A)` (I think a method with no limit is useful)
* `def iterateUntilFixpointAtMost(n: Int)(iteration: A ⇒ A)` (more friendly to `{ .. }` blocks for `iteration`, and if someone wants to put a limit, they already know the value, so no need for an arbitrary default)